project.yaml

```diff
@@ -1463,6 +1499,64 @@ Layer:
     properties:
       minzoom: 16
     advanced: {}
+  - id: "placenames-large-very-low-zoom"
+    name: "placenames-large-very-low-zoom"
+    class: "country state"
+    geometry: "point"
+    <<: *extents
+    Datasource:
+      <<: *osm2pgsql
+      table: |-
+        (SELECT
+            way,
+            way_area/NULLIF(!pixel_width!::real*!pixel_height!::real,0) AS way_pixels,
+            name,
+            ref,
+            admin_level
+          FROM planet_osm_polygon
+          WHERE boundary = 'administrative'
+            AND admin_level = '2'
+            AND name IS NOT NULL
+          ORDER BY admin_level ASC, way_area DESC
+        ) AS placenames_large_very_low_zoom
+    properties:
+      minzoom: 2
+      maxzoom: 3
+    advanced: {}
+  - id: "placenames-large-low-zoom"
+    name: "placenames-large-low-zoom"
+    class: "country state"
+    geometry: "point"
+    <<: *extents
+    Datasource:
+      <<: *osm2pgsql
+      table: |-
+        (SELECT
+            way,
+            way_area/NULLIF(!pixel_width!::real*!pixel_height!::real,0) AS way_pixels,
+            name,
+            ref,
+            admin_level
+          FROM planet_osm_polygon o
+          WHERE boundary = 'administrative'
+            AND (admin_level IN ('0', '1', '2')
+              OR (admin_level IN ('3', '4')
```

Comment:
admin level 0, 1, and 3 labels will disappear on zooming in. I suggest only doing 2 and 4